### ISSUE TITLE
watcher misinterprets "expected to be a file but is a directory", leading to infinite refresh loops (eg, for `meteor --settings /tmp`)

### ISSUE THREAD

**Issue by wackazong** [meteor/meteor#3854]
If you just specify a path and not a file, meteor hangs in preparing the application and neither starts nor logs an error.

**Comment by glasser** [comment]
OK, I assume you mean a directory when you say a path?

We fixed something related to this on devel already with #3757.  But you're right — specifying a directory gives you an infinite refresh.

It's a little better on devel now because it at least prints 

```
=> Modified -- restarting.
=> Errors prevented startup:
   
   While preparing to run:
   /tmp: file not found (settings file)
```

over and over instead of nothing.

**Comment by wackazong** [comment]
Jep, I mean directory.

**Comment by glasser** [comment]
Not sure why you closed it --- it's still a bug, just a slightly better one :)

**Comment by wackazong** [comment]
Ok, I thought you just left it open because you were unsure about the "path" vs "directory" ambiguity.

**Comment by glasser** [comment]
OK, this is a bug in the Meteor 1.0.2 watcher rewrite.  Based on how WatchSets are created, `addFile("path", null)` is supposed to mean "`path` is not a file, though it may be a directory", but then the watcher logic is interpreting it as "`path` does not exist", and thus restarting in a loop.

**Comment by hwillson** [comment]
PR #9030 should address this.

**Comment by hwillson** [comment]
PR #9030 was merged and released in Meteor 1.5.2, so this issue should be resolved. Thanks!
